### PR TITLE
chore(telemetry): Attach anonymous ID of user to the url of the cta link to Atlas VSCODE-249

### PR DESCRIPTION
**Pull request by Anemy**
VSCODE-249

This PR adds the user's telemetry id to the Atlas create cluster link. It's a little hacky using global/attached to `window` js variables, but I couldn't think of a cleaner way to do it easily.

Also a little drive by cleanup on some tests to use async/await and `strictEqual`.

### CODE REVIEW

File: src/test/suite/explorer/databaseTreeItem.test.ts [hunk 1]
```diff
@@ -203,106 +196,94 @@ suite('DatabaseTreeItem Test Suite', () => {
...
-                            testerObjectField.collapsibleState ===
-                            vscode.TreeItemCollapsibleState.Expanded,
-                            `Expected the subdocument field to have an expanded state (2), found ${postCollapseSchemaTreeItem.childrenCache.testerObject.collapsibleState}.`
-                          );
-                        })
-                        .then(done, done);
-                    }, done);
-                }, done);
-            }, done);
-          }, done);
```
Comment (mcasimir) [inline src/test/suite/explorer/databaseTreeItem.test.ts:304]: so nice to see this cascade of callback replaced by async/await :)

File: src/test/suite/views/webview-app/components/atlas-cta/atlas-cta.test.tsx [hunk 2]
```diff
@@ -45,15 +46,16 @@ describe('Resources Panel Component Test Suite', () => {
...
     });
 
     test('when a trusted link is clicked it sends an event to the extension', () => {
       assert(!fakeVscodeWindowPostMessage.called);
+      window[USER_ID_GLOBAL_VARIABLE_NAME] = 'mockUserID';
       wrapper.find('a').at(1).simulate('click');
       assert(fakeVscodeWindowPostMessage.called);
-      assert(fakeVscodeWindowPostMessage.firstCall.args[0].command === 'OPEN_TRUSTED_LINK');
-      assert(fakeVscodeWindowPostMessage.firstCall.args[0].linkTo === 'https://mongodb.com/products/vs-code/vs-code-atlas-signup?utm_campaign=vs-code-extension&utm_source=visual-studio&utm_medium=product');
+      assert.strictEqual(fakeVscodeWindowPostMessage.firstCall.args[0].command, 'OPEN_TRUSTED_LINK');
```
Comment (alenakhineika) [inline src/test/suite/views/webview-app/components/atlas-cta/atlas-cta.test.tsx:57]: The `===` and `strictEqual`do the same, right? Just a different syntax?

File: src/test/suite/explorer/databaseTreeItem.test.ts [hunk 3]
```diff
@@ -150,7 +146,7 @@ suite('DatabaseTreeItem Test Suite', () => {
     );
   });
 
-  test('collections are displayed in the alphanumerical case insensitive order, with system collections last', (done) => {
+  test('collections are displayed in the alphanumerical case insensitive order, with system collections last', async () => {
```
Comment (alenakhineika) [inline src/test/suite/explorer/databaseTreeItem.test.ts:149]: Glad to see these `done` gone!

File: src/test/suite/views/webview-app/components/atlas-cta/atlas-cta.test.tsx (hunk 2 above)
Comment (alenakhineika) [reply src/test/suite/views/webview-app/components/atlas-cta/atlas-cta.test.tsx:57]: Nice! ✨

**Review by mcasimir — APPROVED** [review]
Didn't try it out but looks good!

**Comment by mmarcon** [comment]
@Anemy does this cover the CTA in Help and Feedback too? I quickly scrolled through the PR and I don't see it.

**Comment by Anemy** [comment]
@mmarcon Ah thanks! totally missed that. I'll open a follow up pr.

**Pull request merged**